### PR TITLE
pkg/policy: remove "changed" return value from regenerateConsumable

### PR DESCRIPTION
**Pull request by ianvernon**
regenerateConsumable is only called within regeneratePolicy. The documentation
for regeneratePolicy dictates that it returns whether the BPF for the given
endpoint should be regenerated. However, regenerateConsumable returned \"changed\"
only when the PolicyMap for the endpoint / PolicyMaps for the Consumable were
updated. PolicyMap state changes do not necessitate regeneration of the BPF
program for an endpoint. However, the result of regenerateConsumable, which only
returned \"true\" for when changes were made to PolicyMaps, was used in
determining whether policy was changed in regeneratePolicy, and thus meant that
a change in the PolicyMaps resulted in regeneratePolicy specifying that BPF
needed to be regenerated for the given endpoint. As explained, since BPF
regeneration is not necessary in the case of PolicyMap updates, just remove the
return value for whether regenerateConsumable changed the PolicyMaps for
endpoints when determining whether BPF needs to be regenerated.

Signed-off by: Ian Vernon <ian@cilium.io>

### CODE REVIEW

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

**Comment by ianvernon** [comment]
test-me-please

File: pkg/endpoint/policy.go
Comment (raybejjani) [inline pkg/endpoint/policy.go:314]: is the `_ =` needed for some side effect? I didn't think the compiler was that aggressive about dropping functions.

**Review by raybejjani — APPROVED** [review]

File: pkg/endpoint/policy.go
Comment (ianvernon) [reply pkg/endpoint/policy.go:314]: I prefer specifically indicating that I’m ignoring the result of some functions. Especially because it shows that I am indeed intentionally ignoring the results !

Comment (raybejjani) [reply pkg/endpoint/policy.go:314]: That crossed my mind as the reason. It wasn't apparent because `allowIngressIdentity` returning anything wasn't obvious, nor that the return is important. Anyway, no worries.

Comment (ianvernon) [reply pkg/endpoint/policy.go:314]: Indeed ! This function will be going away soon anyway ;)

**Comment by ianvernon** [comment]
Janitor team was added to review when I had code changes that are no longer part of this PR. Since I have gotten approval from endpoint codeowner, and do not need janitor approval, am merging.

**Pull request merged**
